src/core/src/CoreMain.py

```diff
@@ -101,7 +101,7 @@ def __init__(self, argv):
                     patch_installer.mark_installation_completed()
                     overall_patch_installation_operation_successful = True
                 self.update_patch_substatus_if_pending(patch_operation_requested, overall_patch_installation_operation_successful, patch_assessment_successful, configure_patching_successful, status_handler, composite_logger)
-
+                package_manager.refresh_repo()
```

Comment:
This will not run in a legitimate exception throw from code higher up.

We may need to extend package manager to have refresh_repo_safely() which is just a solid try catch in the PackageManager class over refresh_repo()   -- concrete method, not abstract.

And then refresh_repo_safely can be called at the top of the finally block (requires package_manager to be init to None outside the coremain try catch, and then null check on it in the finally block.

Reply:
Addressed